### ISSUE TITLE
Attribute disapear after changing configurable product

### ISSUE THREAD

**Issue by glwstor** [magento/magento2#13564]
Filterable (with results) attribute for layered navigation will disapear after changing configurable product.

### Preconditions
1. Magento 2.2.2 without sample data
2. Apache 2.4.27
3. PHP 7.1

### Steps to reproduce
Creating a new configurable product with auto generated simple products, the filterable attribute like color, will apear in the category list for layered navigation.
After removing a simple product from the configurable product, the attribute "color" will disapear from the layered navigation.
The attribute will be deleted from the database tabel "catalog_product_entity_int" at the configurable product.
After changing the attribute setting "Use in Layered Navigation" from "with results" to "no results" -> reindex and back to "with results" -> reindex, the attribute will be added to the "catalog_product_entity_int" table.

### Expected result
Filterable attribute should appear on layered navigation

### Actual result
Filterable attribute disapear

![attribut_id](https://user-images.githubusercontent.com/16683372/35973641-6e4492ac-0cd6-11e8-8e3e-4af86e3ea774.jpg)

**Comment by magento-engcom-team** [comment]
Hello @glwstor. Thanks for reporting. Currently we've been not able to reproduce Your issue. Please watch the attached video and tell us where we've got wrong if so.
[Conf_attr.zip](https://github.com/magento/magento2/files/1710953/Conf_attr.zip)

**Comment by glwstor** [comment]
Hello @magento-engcom-team, attached you will find the video.
[configurable-attribute-issue.zip](https://github.com/magento/magento2/files/1715769/configurable-attribute-issue.zip)

**Comment by glwstor** [comment]
Hello @magento-engcom-team, Attached you will find another strange behavior with layered navigation.
After reindexing the filtarable attrbiute will disapear partially.
This is tested on a complete fresh installed magento2.2.2. Inventory is disabled.
[fresh-magento2.2.2-configurable-issue.zip](https://github.com/magento/magento2/files/1715877/fresh-magento2.2.2-configurable-issue.zip)

**Comment by glwstor** [comment]
If every configurable product has the same colors like white,blue,green,red, than all colors will appear in the layered navigation.
If one color like "green" of a configurable product is missing, than only the missing color will appear in the layered navigation. This is only after reindexing. See item SKU Test3
![config_color_one_missing](https://user-images.githubusercontent.com/16683372/36096523-09e95bac-0ff7-11e8-93bc-04f116620f21.jpg)
![config_color](https://user-images.githubusercontent.com/16683372/36096525-0b6a8438-0ff7-11e8-9981-f95e33cb7ee4.jpg)

**Comment by magento-engcom-team** [comment]
@glwstor, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by magento-engcom-team** [comment]
@glwstor, thank you for your report.     
    
Unfortunately, we are archiving this ticket now as it did not get much attention from both Magento Community and Core developers for an extended period. This is done in an effort to create a quality, community-driven backlog which will allow us to allocate the required attention more easily.

Please feel free to comment, reopen or create new ticket according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) 
if you are still facing this issue on the latest `2.4-develop` branch. Thank you for collaboration.
